python/dataset.cpp

```diff
@@ -312,35 +318,32 @@ void init_dataset(py::module &m) {
   bind_helper_view<items_view, DatasetView>(m, "DatasetView");
   bind_helper_view<items_view, CoordsView>(m, "CoordsView");
   bind_helper_view<items_view, MasksView>(m, "MasksView");
-  bind_helper_view<items_view, AttrsView>(m, "AttrsView");
   bind_helper_view<keys_view, Dataset>(m, "Dataset");
   bind_helper_view<keys_view, DatasetView>(m, "DatasetView");
   bind_helper_view<keys_view, CoordsView>(m, "CoordsView");
   bind_helper_view<keys_view, MasksView>(m, "MasksView");
-  bind_helper_view<keys_view, AttrsView>(m, "AttrsView");
   bind_helper_view<values_view, Dataset>(m, "Dataset");
   bind_helper_view<values_view, DatasetView>(m, "DatasetView");
   bind_helper_view<values_view, CoordsView>(m, "CoordsView");
   bind_helper_view<values_view, MasksView>(m, "MasksView");
-  bind_helper_view<values_view, AttrsView>(m, "AttrsView");
 
   bind_mutable_view<CoordsView, CoordsConstView>(m, "Coords");
   bind_mutable_view<MasksView, MasksConstView>(m, "Masks");
-  bind_mutable_view<AttrsView, AttrsConstView>(m, "Attrs");
 
   py::class_<DataArray> dataArray(m, "DataArray", R"(
     Named variable with associated coords, masks, and attributes.)");
   dataArray.def(py::init<const DataArrayConstView &>());
-  dataArray.def(py::init([](VariableConstView data,
-                            std::map<Dim, VariableConstView> coords,
-                            std::map<std::string, VariableConstView> masks,
-                            std::map<std::string, VariableConstView> attrs) {
-                  return DataArray{Variable{data}, coords, masks, attrs};
-                }),
-                py::arg("data") = Variable{},
-                py::arg("coords") = std::map<Dim, VariableConstView>{},
-                py::arg("masks") = std::map<std::string, VariableConstView>{},
-                py::arg("attrs") = std::map<std::string, VariableConstView>{});
+  dataArray.def(
+      py::init([](VariableConstView data,
+                  std::map<Dim, VariableConstView> coords,
+                  std::map<std::string, VariableConstView> masks,
+                  std::map<Dim, VariableConstView> unaligned_coords) {
+        return DataArray{Variable{data}, coords, masks, unaligned_coords};
+      }),
+      py::arg("data") = Variable{},
+      py::arg("coords") = std::map<Dim, VariableConstView>{},
```

Comment:
DataArray itself provides three accessors (looking at c++ class), unaligned_coords, aligned_coords, coords (both), but I think here you are using the "old" constructor argument "coords" to refer to the aligned coordinates. Should this be updated?

Reply:
This is intentional. It ensures that in most cases we can just use the consise `coords`, and also ensures a symmetric API with `Dataset`. `DataArray.coords` indeed gives both aligned and unaligned, but inserting into it adds aligned coords, so the property is also consistent with the keyword argument to the constructor.

Reply:
Ah yes, I think we did cover the symmetric argument during our walk through.